test/storage-luatest/storage_1_1_1_test.lua

```diff
@@ -0,0 +1,102 @@
+local t = require('luatest')
+local vtest = require('test.luatest_helpers.vtest')
+local vutil = require('vshard.util')
+
+local group_config = {{engine = 'memtx'}, {engine = 'vinyl'}}
+
+if vutil.feature.memtx_mvcc then
+    table.insert(group_config, {
+        engine = 'memtx', memtx_use_mvcc_engine = true
+    })
+    table.insert(group_config, {
+        engine = 'vinyl', memtx_use_mvcc_engine = true
+    })
+end
+
+local test_group = t.group('storage', group_config)
+
+local cfg_template = {
+    sharding = {
+        {
+            replicas = {
+                replica_1_a = {
+                    master = true,
+                },
+            },
+        },
+        {
+            replicas = {
+                replica_2_a = {
+                    master = true,
+                },
+            },
+        },
+        {
+            replicas = {
+                replica_3_a = {
+                    master = true,
+                },
+            },
+        },
+    },
+    bucket_count = 15
+}
+local global_cfg
+
+test_group.before_all(function(g)
+    cfg_template.memtx_use_mvcc_engine = g.params.memtx_use_mvcc_engine
+    global_cfg = vtest.config_new(cfg_template)
+
+    vtest.cluster_new(g, global_cfg)
+    vtest.cluster_bootstrap(g, global_cfg)
+    vtest.cluster_rebalancer_disable(g)
+end)
+
+test_group.after_all(function(g)
+    g.cluster:drop()
+end)
+
+--
+-- Test that manual vshard.storage.bucket_send() cannot lead to
+-- doubled buckets (gh-414).
+--
+test_group.test_manual_bucket_send_doubled_buckets = function(g)
+    vtest.cluster_exec_each_master(g, function()
+        _G.bucket_recovery_pause()
+    end)
+
+    local uuid_2 = g.replica_2_a:exec(function()
+        ivshard.storage.internal.errinj.ERRINJ_LONG_RECEIVE = true
+        return box.info.cluster.uuid
```

Comment:
This won't work in 3.0 - you need to rebase on latest vshard master and use `util.replicaset_uuid()`.